src/Confluent.Kafka/AdminClient.cs

```diff
@@ -0,0 +1,750 @@
+                {
+                    Topic = PtrToStringUTF8(Librdkafka.topic_result_name(topicResultPtr)),
+                    Error = new Error(
+                        Librdkafka.topic_result_error(topicResultPtr), 
+                        PtrToStringUTF8(Librdkafka.topic_result_error_string(topicResultPtr)))
+                }).ToList();
+        }
+
+        private ConfigEntryResult extractConfigEntry(IntPtr configEntryPtr)
+        {
+            var synonyms = new List<ConfigSynonym>();
+            var synonymsPtr = Librdkafka.ConfigEntry_synonyms(configEntryPtr, out UIntPtr synonymsCount);
+            if (synonymsPtr != IntPtr.Zero)
+            {
+                IntPtr[] synonymsPtrArr = new IntPtr[(int)synonymsCount];
+                Marshal.Copy(synonymsPtr, synonymsPtrArr, 0, (int)synonymsCount);
+                synonyms = synonymsPtrArr
+                    .Select(synonymPtr => extractConfigEntry(synonymPtr))
+                    .Select(e => new ConfigSynonym { Name = e.Name, Value = e.Value, Source = e.Source } )
+                    .ToList();
+            }
+
+            return new ConfigEntryResult
+            {
+                Name = PtrToStringUTF8(Librdkafka.ConfigEntry_name(configEntryPtr)),
+                Value = PtrToStringUTF8(Librdkafka.ConfigEntry_value(configEntryPtr)),
+                IsDefault = (int)Librdkafka.ConfigEntry_is_default(configEntryPtr) == 1,
+                IsSensitive = (int)Librdkafka.ConfigEntry_is_sensitive(configEntryPtr) == 1,
+                IsReadOnly = (int)Librdkafka.ConfigEntry_is_read_only(configEntryPtr) == 1,
+                Source = Librdkafka.ConfigEntry_source(configEntryPtr),
+                Synonyms = synonyms
+            };
+        }
+
+        private List<DescribeConfigResult> extractResultConfigs(IntPtr configResourcesPtr, int configResourceCount)
+        {
+            var result = new List<DescribeConfigResult>();
+
+            IntPtr[] configResourcesPtrArr = new IntPtr[configResourceCount];
+            Marshal.Copy(configResourcesPtr, configResourcesPtrArr, 0, configResourceCount);
+            foreach (var configResourcePtr in configResourcesPtrArr)
+            {
+                var resourceName = PtrToStringUTF8(Librdkafka.ConfigResource_name(configResourcePtr));
+                var errorCode = Librdkafka.ConfigResource_error(configResourcePtr);
+                var errorReason = PtrToStringUTF8(Librdkafka.ConfigResource_error_string(configResourcePtr));
+                var resourceConfigType = Librdkafka.ConfigResource_type(configResourcePtr);
+
+                var configEntriesPtr = Librdkafka.ConfigResource_configs(configResourcePtr, out UIntPtr configEntryCount);
+                IntPtr[] configEntriesPtrArr = new IntPtr[(int)configEntryCount];
+                if ((int)configEntryCount > 0)
+                {
+                    Marshal.Copy(configEntriesPtr, configEntriesPtrArr, 0, (int)configEntryCount);
```

Comment:
this is old-skool. I could use `Span<IntPtr>` here but i don't think this is worth bothering to fix up.

Reply:
what I have is definitely safe and it's not performance critical here. I want to leave as is rather than research what's the perfect way to do this or leave a comment that admits my lack of knowledge.